qmf/src/libraries/qmfclient/qmailmessage.cpp

```diff
@@ -4169,7 +4179,11 @@ QByteArray QMailMessagePartContainerPrivate::boundary() const
 
 void QMailMessagePartContainerPrivate::setBoundary(const QByteArray& text)
 {
-    _boundary = text;
+    if (text == QByteArray("NIL")) {
```

Comment:
I think this check for NIL should be made in the BODYSTRUCTURE handling code, before calling setBoundary().  "NIL" could be a valid boundary string in a correctly formatted message.